service/state/core.go

```diff
@@ -232,7 +176,7 @@ func (c *Core) payload(height uint64, path ledger.Path) (*ledger.Payload, error)
 			return dps.ErrNotFound
 		}
 		err := it.Item().Value(func(val []byte) error {
-			val, err := c.decompressor.DecodeAll(val, nil)
+			val, err := decoder.DecodeAll(val, nil)
```

Comment:
I don't really understand why you handle this differently from other instances. IMO, this should simply be one call to `SaveDelta(path, height, delta)`.